pkg/collect/logs.go

```diff
@@ -46,66 +46,97 @@ func (c *CollectLogs) Collect(progressChan chan<- interface{}) (CollectorResult,
+					for _, container := range pod.Spec.Containers {
+						containerNames = append(containerNames, container.Name)
+					}
+					for _, container := range pod.Spec.InitContainers {
+						containerNames = append(containerNames, container.Name)
+					}
 
-				for _, containerName := range containerNames {
-					podLogs, err := savePodLogs(ctx, c.BundlePath, client, &pod, c.Collector.Name, containerName, c.Collector.Limits, false, true)
-					if err != nil {
-						key := fmt.Sprintf("%s/%s-errors.json", c.Collector.Name, pod.Name)
-						if containerName != "" {
-							key = fmt.Sprintf("%s/%s/%s-errors.json", c.Collector.Name, pod.Name, containerName)
-						}
-						err := output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+					for _, containerName := range containerNames {
+						podLogs, err := savePodLogs(ctx, c.BundlePath, client, &pod, c.Collector.Name, containerName, c.Collector.Limits, false, true)
 						if err != nil {
-							return nil, err
+							key := fmt.Sprintf("%s/%s-errors.json", c.Collector.Name, pod.Name)
+							if containerName != "" {
+								key = fmt.Sprintf("%s/%s/%s-errors.json", c.Collector.Name, pod.Name, containerName)
+							}
+							err := output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+							if err != nil {
+								errCh <- err
+							}
+							continue
 						}
-						continue
-					}
-					for k, v := range podLogs {
-						output[k] = v
+						for k, v := range podLogs {
+							output[k] = v
+
+						}
+						resultCh <- output
 					}
-				}
-			} else {
-				for _, container := range c.Collector.ContainerNames {
-					containerLogs, err := savePodLogs(ctx, c.BundlePath, client, &pod, c.Collector.Name, container, c.Collector.Limits, false, true)
-					if err != nil {
-						key := fmt.Sprintf("%s/%s/%s-errors.json", c.Collector.Name, pod.Name, container)
-						err := output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+				} else {
+					for _, container := range c.Collector.ContainerNames {
+						containerLogs, err := savePodLogs(ctx, c.BundlePath, client, &pod, c.Collector.Name, container, c.Collector.Limits, false, true)
 						if err != nil {
-							return nil, err
+							key := fmt.Sprintf("%s/%s/%s-errors.json", c.Collector.Name, pod.Name, container)
+							err := output.SaveResult(c.BundlePath, key, marshalErrors([]string{err.Error()}))
+							if err != nil {
+								errCh <- err
+							}
+							continue
 						}
-						continue
-					}
-					for k, v := range containerLogs {
-						output[k] = v
+						for k, v := range containerLogs {
+							output[k] = v
+						}
+						resultCh <- output
 					}
 				}
 			}
+
 		}
-	}
 
+	}()
+
+	select {
+	case <-ctxTimeout.Done():
+		return nil, errors.New("context timeout exceeded")
+	case o := <-resultCh:
+		output = o
+	case err := <-errCh:
+		if errors.Is(err, context.DeadlineExceeded) {
+			return nil, err
+		}
+		return output, err
```

Comment:
I think this should just return `nil, err` so as not to change the previous behaviour. If there is an error, all the logs were being discarded.

Reply:
fixed code base on recommendations.